site/utils/formUtils.ts

```diff
@@ -98,26 +100,40 @@ export const getStopType = (stopType: string | undefined) => {
     }
 };
 
-// eslint-disable-next-line @typescript-eslint/require-await
-export const filterServices = async (servicesData?: Service[]) => {
-    let services: Service[] = [];
+export const filterServices = (servicesData?: ServiceApiResponse[]) => {
+    let services: ServiceApiResponse[] = [];
+
     if (servicesData && servicesData.length > 0) {
         services = sortServices(servicesData);
+        const filterKey = services[0].dataSource === "tnds" ? "serviceCode" : "lineId";
+        return removeDuplicateServicesByKey(services, filterKey, currentDate);
+    }
+    return services;
+};
 
-        const setOfServices = new Set();
-
-        const filteredServices: Service[] = services.filter((item) => {
-            const serviceDisplay = item.lineName + item.origin + item.destination + item.operatorShortName;
-            if (!setOfServices.has(serviceDisplay)) {
-                setOfServices.add(serviceDisplay);
-                return true;
-            } else {
-                return false;
-            }
-        });
+const currentDate = getDate();
```

Comment:
We can move this into the `removeDuplicateServicesByKey` function, then we don't need to pass it between the functions, it'll allow the mockdate to work as well

Reply:
fixed